app/vmui/packages/vmui/src/components/CustomPanel/NestedNav/NestedNav.tsx

```diff
@@ -0,0 +1,83 @@
+import {TracingData} from "../../../api/types";
+import Box from "@mui/material/Box";
+import ListItem from "@mui/material/ListItem";
+import ListItemText from "@mui/material/ListItemText";
+import ListItemButton from "@mui/material/ListItemButton";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ExpandLess from "@mui/icons-material/ExpandLess";
+import AddCircleRoundedIcon from "@mui/icons-material/AddCircleRounded";
+import Collapse from "@mui/material/Collapse";
+import List from "@mui/material/List";
+import React from "preact/compat";
+import {BorderLinearProgressWithLabel} from "../../BorderLineProgress/BorderLinearProgress";
+
+interface RecursiveProps {
+  tracingData: TracingData[];
+  totalMicrosec: number;
+  openLevels: Record<number, boolean>;
+  onChange: (level: number) => void;
+}
+
+type RecursiveComponent = (props: RecursiveProps) => JSX.Element;
+
+export const recursiveComponent: RecursiveComponent = ({ tracingData, openLevels, totalMicrosec, onChange})  => {
+  const handleListClick = (level: number) => () => onChange(level);
+  const renderData = tracingData.sort((a, b) => {
```

Comment:
Trace entries must be shown in the order they appear in the response, since they usually reflect **sequential** stages of query execution (the only exception is parallel execution of some stages). It looks like this sorting breaks this assumption.

Reply:
removed sorting